### DISCUSSION TITLE
Router disconnects after power failure and needs to be repaired

### DISCUSSION THREAD

**Discussion by ptr727** [tube0013/tube_gateways#100] [Q&A]
I've now twice noticed that after a power failure the tubes router fails to show up on the network, and needs to be repaired.
I use a tubes 2652P2 coordinator and a tubes router as dedicated router, along with several other powered devices acting as routers.

Two days ago we had a short power failure, yesterday I noticed the Z2M docker would not start because it failed to connect to the tubes coordinator.
Restarted the coordinator, z2m started, devices started coming back online.
This morning the only outstanding device is the tubes router.

Before I repair it again, is there some other troubleshooting or remediation I can do to make the coordinator and router more resilient?

**Comment by tube0013** [comment]
I will need to do some testing. I won't be able to do that until later this evening ~10 hours from now.

**Comment by tube0013** [comment]
Apologies I didn't get to this last night, and will try again tonight.

**Comment by tube0013** [comment]
I haven't been able to re-produce this. I have a router on a circuit that I was working on yesterday and the circuit was toggled many times trough out the day and router has come back fine. Will still keep any eye on it.

**Comment by ptr727** [comment]
It seemed sporadic at best, will just repair again.
As before I enable pairing mode in Z2M and pross the BSL button and it paired back up.

**Comment by ptr727** [comment]
Just happened again, but this time not a power event, I unplugged the coordinator LAN cable (keeping power on) to move it and plugged LAN cable back in, router went offline.


```
Warning 2022-11-04 09:57:27Failed to ping 'Tubes-Router-8F1717' (attempt 1/1, Read 0x00124b00258f1717/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 57150 - 1 - 10 - 0 - 1 after 10000ms))
```

**Comment by tube0013** [comment]
I got nothing. You can try the latest Dev z-stack FW , or roll back to the last Feb release of either Z-Stack on my version (functionally the same) https://github.com/tube0013/tube_gateways/tree/main/models/current/tubeszb-router/firmware

**Comment by ptr727** [comment]
What seems weird is that Z2M reports the router as offline, but it also reports last seen 28 minutes ago?

```
Warning 2022-11-04 17:43:56Failed to ping 'Tubes-Router-8F1717' (attempt 1/1, Read 0x00124b00258f1717/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 57150 - 1 - 220 - 0 - 1 after 10000ms))
```

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/2061579/200093648-589dbb36-8176-4fb6-b381-ba30ebfbb32c.png">

I am currently running `tubeszb_router_20220223.hex`, I will try `tubeszb_router_20221002.hex.zip`.

**Reply by tube0013** [reply]
Someone else reported the router showing offline in z2m but was able to "ping" it. I think there is something possibly going on with how it determines its offline. If you can read a cluster it's most definitely online.

**Reply by ptr727** [reply]
I just noticed something similar, as I was writing a bug report for z2m, the router reported back online.
See: https://github.com/Koenkk/zigbee2mqtt/issues/14832
